### PR TITLE
Eng 943 update getartifactversionsroute to include checks

### PR DESCRIPTION
**Pull request by eunice-chan**
Associated UI Task: ENG-918 Implement Artifact details page

Running `curl --header "api-key: NH264SBCGTV91RXDJOYLKQ78M0PAZIEW" http://3.142.105.48:8080/api/artifact_versions` gives me:
```
{
   "latest_versions":{
      "02db8dc3-920b-48f0-ad43-a0e80b271368":{
         "workflow_name":"complex",
         "artifact_name":"another_dffn artifact",
         "workflow_id":"c42c2df0-9bc7-4066-b4a6-50a289a7f779",
         "load_specs":[
            {
               "service":"SQLite",
               "integration_id":"0aa4f450-ef07-47ff-889e-633a1d77b64f",
               "parameters":{
                  "table":"df2",
                  "update_mode":"fail"
               }
            }
         ],
         "versions":{
            "1e5e52d1-5167-4900-958c-72423b37ada3":{
               "timestamp":1654824799,
               "status":"succeeded",
               "error":"",
               "checks":null
            }
         }
      },
      "1f6b8565-889d-455d-b14a-39f54817e340":{
         "workflow_name":"complex3",
         "artifact_name":"another_dffn artifact",
         "workflow_id":"334be0a3-d845-47a5-8e4a-3d8307618a75",
         "load_specs":[
            {
               "service":"SQLite",
               "integration_id":"0aa4f450-ef07-47ff-889e-633a1d77b64f",
               "parameters":{
                  "table":"df2",
                  "update_mode":"fail"
               }
            }
         ],
         "versions":{
            "1bdcec8f-c23e-41c7-af71-503ca702902b":{
               "timestamp":1654824899,
               "status":"succeeded",
               "error":"",
               "checks":null
            }
         }
      },
      "6658a656-498d-4f46-9d1b-64356257105f":{
         "workflow_name":"checks_workflow_5",
         "artifact_name":"op_fn artifact",
         "workflow_id":"59065b1a-968a-442a-b572-1926731234dc",
         "load_specs":[
            {
               "service":"SQLite",
               "integration_id":"0aa4f450-ef07-47ff-889e-633a1d77b64f",
               "parameters":{
                  "table":"op_df",
                  "update_mode":"fail"
               }
            }
         ],
         "versions":{
            "a5fa8e61-c78a-446a-9b5e-c2b55b476f32":{
               "timestamp":1654895553,
               "status":"succeeded",
               "error":"",
               "checks":null
            },
            "d3c014a4-e275-4b56-ae61-5e371f8c6ae5":{
               "timestamp":1654824648,
               "status":"succeeded",
               "error":"",
               "checks":null
            }
         }
      },
      "cebe10f5-41ce-4756-bec4-c79f23922200":{
         "workflow_name":"complex4",
         "artifact_name":"another_dffn artifact",
         "workflow_id":"8e60d3a3-a6b9-4e94-8c9d-440d010f9a26",
         "load_specs":[
            {
               "service":"SQLite",
               "integration_id":"0aa4f450-ef07-47ff-889e-633a1d77b64f",
               "parameters":{
                  "table":"df2",
                  "update_mode":"fail"
               }
            }
         ],
         "versions":{
            "a3cd7769-14c6-4e8b-8607-19aae256dfaa":{
               "timestamp":1654825880,
               "status":"succeeded",
               "error":"",
               "checks":[
                  {
                     "name":"check_op_2",
                     "status":"succeeded",
                     "metadata":{
                        "error":"",
                        "logs":{
                           
                        }
                     }
                  }
               ]
            }
         }
      },
      "d00751f8-022b-44c5-a848-5e0c8b9b26d5":{
         "workflow_name":"complex2",
         "artifact_name":"another_dffn artifact",
         "workflow_id":"731a3442-8b36-4a61-99d8-34381e4590d6",
         "load_specs":[
            {
               "service":"SQLite",
               "integration_id":"0aa4f450-ef07-47ff-889e-633a1d77b64f",
               "parameters":{
                  "table":"df2",
                  "update_mode":"fail"
               }
            }
         ],
         "versions":{
            "f8401d0a-1906-4096-9fea-52443ff70976":{
               "timestamp":1654824809,
               "status":"succeeded",
               "error":"",
               "checks":null
            }
         }
      },
      "d62dd224-f669-479c-b123-9d56c8e54e62":{
         "workflow_name":"checks_workflow_4",
         "artifact_name":"op_fn artifact",
         "workflow_id":"5b9ce7ef-2577-47d2-90af-1c4d8de15e08",
         "load_specs":[
            {
               "service":"SQLite",
               "integration_id":"0aa4f450-ef07-47ff-889e-633a1d77b64f",
               "parameters":{
                  "table":"df",
                  "update_mode":"fail"
               }
            }
         ],
         "versions":{
            "ceb0fcdc-29ac-4e53-89de-992fdafb016c":{
               "timestamp":1654824625,
               "status":"succeeded",
               "error":"",
               "checks":null
            }
         }
      }
   },
   "historical_versions":{
      "fde9c4f7-894b-4cfd-8c8d-9b77aa0a5e93":{
         "workflow_name":"complex",
         "artifact_name":"",
         "workflow_id":"00000000-0000-0000-0000-000000000000",
         "load_specs":[
            {
               "service":"SQLite",
               "integration_id":"0aa4f450-ef07-47ff-889e-633a1d77b64f",
               "parameters":{
                  "table":"df",
                  "update_mode":"fail"
               }
            }
         ],
         "versions":{
            "8e0527a3-a3bf-46c7-8bb0-438594aabfe4":{
               "timestamp":1654824780,
               "status":"succeeded",
               "error":"",
               "checks":null
            }
         }
      }
   }
}
```
If there are no checks, the version looks like this:
```
            "8e0527a3-a3bf-46c7-8bb0-438594aabfe4":{
               "timestamp":1654824780,
               "status":"succeeded",
               "error":"",
               "checks":null
            }
```
If there are checks, the version looks like this:
```
            "a3cd7769-14c6-4e8b-8607-19aae256dfaa":{
               "timestamp":1654825880,
               "status":"succeeded",
               "error":"",
               "checks":[
                  {
                     "name":"check_op_2",
                     "status":"succeeded",
                     "metadata":{
                        "error":"",
                        "logs":{
                           
                        }
                     }
                  }
               ]
            }
```

### CODE REVIEW

**Comment by likawind** [comment]
I assume there will be UI (and maybe SDK) changes to follow up?

**Comment by eunice-chan** [comment]
@likawind Yes this will be used in the Data Artifacts Details Page, which hasn't been created yet.

File: src/golang/cmd/server/queries/standard.go [hunk 1]
```diff
@@ -99,6 +99,29 @@ func (r *standardReaderImpl) GetArtifactResultsByArtifactIds(
...
 
+func (r *standardReaderImpl) GetCheckResultsByArtifactIds(
+	ctx context.Context,
+	artifactIds []uuid.UUID,
+	db database.Database,
+) ([]ArtifactCheckResponse, error) {
+	if len(artifactIds) == 0 {
+		return nil, errors.New("Provided empty IDs list.")
+	}
+	query := fmt.Sprintf(
```
Comment (cw75) [inline src/golang/cmd/server/queries/standard.go:110]: Add more spacing. You can refer to the function above for an example.

File: src/golang/cmd/server/handler/get_artifact_versions.go [hunk 2]
```diff
@@ -199,6 +208,31 @@ func (h *GetArtifactVersionsHandler) Perform(ctx context.Context, interfaceArgs
 		failedArtifactIds = append(failedArtifactIds, failedArtifactId)
 	}
 
+	checkResults, err := h.CustomReader.GetCheckResultsByArtifactIds(ctx, allArtifactIds, h.Database)
+	if err != nil {
+		return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to get artifact versions.")
+	}
+
+	// We now fill in the validation test result. We can join the validation test result with the correct
+	// artifact version by the workflow dag result id.
```
Comment (cw75) [inline src/golang/cmd/server/handler/get_artifact_versions.go:217]: change "validation" to "check"

**Review by cw75 — APPROVED** [review]
Left a couple comments about styles and nits, but otherwise LGTM.

**Review by likawind — APPROVED** [review]
Changes LGTM. Besides Chenggang's suggestions, would you mind updating the PR description to include a pointer to the UI PR as well?

**Pull request merged**